### ISSUE TITLE
ConfigManager set float

### ISSUE THREAD

**Issue by mvolotov** [Redempt/RedLib#20]
Hi, I ran into a problem while using Config Manager. I have a registered map of the format <String, Locations>, where Locations is an object of my class containing information about the location and additional information. Everything works fine, except saving numbers in float format. In the config file, they are saved in the double format and cannot be loaded further due to a type mismatch.

Example:
```
Locations class - who has a float member yaw and pitch.
@ConfigMappable
public class Locations {
        @ConfigValue
        private String world;

        @ConfigValue
        private double x;

        @ConfigValue
        private double y;

        @ConfigValue
        private double z;

        @ConfigValue
        private float yaw;

        @ConfigValue
        private float pitch;

        protected Locations() {}

        public Locations(Location loc){
                this.world = loc.getWorld().getName();
                this.x = loc.getX();
                this.y = loc.getY();
                this.z = loc.getZ();
                this.yaw = loc.getYaw();
                this.pitch = loc.getPitch();
        }
}

```
Saved config, and get this values in file.
```
locationMap:
  sample:
    world: world
    x: -1982.699999988079
    y: 80.0
    z: -302.1308459618936
    yaw: 42.650646
    pitch: 5.249998
```
Get this java exception, when try load this values from config.
```

[12:27:26 ERROR]: Error occurred while enabling CustomRWTP v1.0-SNAPSHOT (Is it up to date?)
java.lang.IllegalArgumentException: Can not set float field org.asteriagame.customrwtp.config.Locations.yaw to java.lang.Double
at jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(Unknown Source) ~[?:?]
at jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(Unknown Source) ~[?:?]
at jdk.internal.reflect.UnsafeFloatFieldAccessorImpl.set(Unknown Source) ~[?:?]
at java.lang.reflect.Field.set(Unknown Source) ~[?:?]
at redempt.redlib.configmanager.ConfigField.load(ConfigField.java:72) ~[?:?]
at redempt.redlib.configmanager.ConfigObjectMapper.lambda$loadMapped$1(ConfigObjectMapper.java:93) ~[?:?]
at java.util.ArrayList.forEach(Unknown Source) ~[?:?]
at redempt.redlib.configmanager.ConfigObjectMapper.loadMapped(ConfigObjectMapper.java:93) ~[?:?]
at redempt.redlib.configmanager.ConfigObjectMapper.load(ConfigObjectMapper.java:79) ~[?:?]
at redempt.redlib.configmanager.ConfigMap.lambda$load$0(ConfigMap.java:37) ~[?:?]
at java.lang.Iterable.forEach(Unknown Source) ~[?:?]
at redempt.redlib.configmanager.ConfigMap.load(ConfigMap.java:36) ~[?:?]
at redempt.redlib.configmanager.ConfigField.load(ConfigField.java:48) ~[?:?]
at redempt.redlib.configmanager.ConfigManager.lambda$load$4(ConfigManager.java:288) ~[?:?]
at java.util.ArrayList.forEach(Unknown Source) ~[?:?]
at redempt.redlib.configmanager.ConfigManager.lambda$load$5(ConfigManager.java:288) ~[?:?]
at java.util.HashMap.forEach(Unknown Source) ~[?:?]
at redempt.redlib.configmanager.ConfigManager.load(ConfigManager.java:287) ~[?:?]
at org.asteriagame.customrwtp.CustomRWTP.onEnable(CustomRWTP.java:21) ~[?:?]
at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.5.jar:git-Paper-786]
at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.16.5.jar:git-Paper-786]
at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.16.5.jar:git-Paper-786]
at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:518) ~[patched_1.16.5.jar:git-Paper-786]
at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:432) ~[patched_1.16.5.jar:git-Paper-786]
at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:594) ~[patched_1.16.5.jar:git-Paper-786]
at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:298) ~[patched_1.16.5.jar:git-Paper-786]
at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1069) ~[patched_1.16.5.jar:git-Paper-786]
at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Paper-786]
at java.lang.Thread.run(Unknown Source) [?:?]
```

P.S. sry for my bad English

**Comment by UntouchedOdin0** [comment]
`java.lang.IllegalArgumentException: Can not set float field org.asteriagame.customrwtp.config.Locations.yaw to java.lang.Double
`
It looks like you're trying to set a Double value to a float value.

**Comment by boxbeam** [comment]
Fixed with release 6.0.1, thank you for reporting this bug
